tests/hikari/test_snowflake.py

```diff
@@ -120,3 +122,21 @@ class NekoUnique2(snowflakes.Unique):
         assert unique1 == unique2
         assert unique1 != NekoUnique2()
         assert unique1 != raw_id
+
+
+@pytest.mark.parametrize(
+    ("guild_id", "expected_id"),
+    [(140502780547694592, 2), ("655288690192416778", 1), (snowflakes.Snowflake(105785483455418368), 3)],
+)
+@pytest.mark.parametrize("shard_count", [4, "4"])
+def test_calculate_shard_id_with_shard_count(guild_id, expected_id, shard_count):
+    assert snowflakes.calculate_shard_id(shard_count, guild_id) == expected_id
+
+
+@pytest.mark.parametrize(
+    ("guild_id", "expected_id"),
+    [(140502780547694592, 2), ("115590097100865541", 5), (snowflakes.Snowflake(105785483455418368), 7)],
+)
+def test_calculate_shard_id_with_app(guild_id, expected_id):
+    mock_app = mock.Mock(bot.BotApp, shard_count=8)
+    assert snowflakes.calculate_shard_id(mock_app, guild_id) == expected_id
```

Comment:
what if it isn't a bot app? (i.e. called from a model that might be on a rest-only app)

Reply:
it's typed as taking ShardAware and if they gave it the wrong type it'd just error as expected